cpg-core/src/test/resources/function-dfg.json

```diff
@@ -0,0 +1,56 @@
+[
+  {
+    "functionDeclaration": {
+      "language": "de.fraunhofer.aisec.cpg.frontends.java.JavaLanguage",
+      "methodName": "java.util.List.addAll",
+      "signature": ["int", "java.util.Object"]
+    },
+    "dataFlows": [
+      {
+        "from": "param1",
+        "to": "base",
+        "dfgType": "full"
```

Comment:
Using Json to then specify partial DFGs is than not possible correct? because we cannot specify the target as string? Or can we just specify partial here and what expression is associated can be inferred?

Reply:
In theory we could use the shortcut API to lookup a node using a name for this, but it gets complicated when there are multiple nodes with the same name, but different types. I would keep this as future work.

Reply:
The node that would be referenced in that string is somewhere along the DFG leading to that call, so it can be identified more easily i think, but we can do that at some future point as you said.

Reply:
I'm not sure if it's easy to identify a node based on the path in all cases. Most likely, the "partial" DFG edge would be annotated with a `FieldDeclaration`, right? Since that one should have a (unique) FQN, we could use that name here.

Another use-case for this property could be to pass some index for a return value like a tuple or multi-value return instead of using the `returnX` thing I'm using right now. I'm not completely sure if/how we want to note all these things though.

I'd like to finalize the JSON/YAML format within this PR as good as we can, so if you have ideas/wishes on other scenarios and on how to note them, it would be great to add this here.

Reply:
ok, we keep it for the future